addons/web/static/src/js/views/list/list_controller.js

```diff
@@ -99,6 +99,24 @@ var ListController = BasicController.extend({
             return record.res_id;
         });
     },
+
+    /**
+     * Returns the list of currently selected res_ids (with the check boxes on
+     * the left) or the whole domain res_ids if it is selected
+     *
+     * This method should be the implementation of getSelectedIds but it is kept for compatibility reasons
+     *
+     * @returns {Promise<integer[]>}
+     */
+    getSelectedIdsWithDomain: async function () {
+        if (this.isDomainSelected) {
+            const state = this.model.get(this.handle, {raw: true});
+            return await this._domainToResIds(state.getDomain(), session.active_ids_limit);
+        } else {
+            return Promise.resolve(this.model.localIdsToResIds(this.selectedRecords));
```

Comment:
I returned a Promise here in case someone wants to call then on the result

Reply:
:+1: